test/parallel/test-zlib-dictionary.js

```diff
@@ -69,6 +69,66 @@ function run(num) {
 }
 run(1);
 
+function rawDictionaryTest() {
+  let output = '';
+  const deflate = zlib.createDeflateRaw({ dictionary: spdyDict });
+  const inflate = zlib.createInflateRaw({ dictionary: spdyDict });
+
+  deflate.on('data', function(chunk) {
+    inflate.write(chunk);
+  });
+
+  inflate.on('data', function(chunk) {
+    output += chunk;
+  });
+
+  deflate.on('end', function() {
+    inflate.end();
+  });
+
+  inflate.on('end', common.mustCall(function() {
+    assert.equal(input, output);
```

Comment:
Seeing how different the tests are here I imagine backporting the other one would be quite painful, yes, I don't think I'm masochistic enough for that. I can address the points you raise if you like, but they were not part of the [original first commit](https://github.com/nodejs/node/pull/8512/commits/faf96113150ae617af4ab0931a9c807705c5b4c8) either, so I skipped them to keep the same style as the surrounding code.